azurecompute-arm/src/main/java/org/jclouds/azurecompute/arm/compute/functions/LocationToLocation.java

```diff
@@ -47,20 +47,14 @@
    @Override
    public org.jclouds.domain.Location apply(final Location location) {
       final LocationBuilder builder = new LocationBuilder();
-      String id = location.id();
-      int index = id.lastIndexOf('/');
-      if (index > 0 && (index + 1) < id.length())
-         id = id.substring(index + 1);
-      builder.id(id);
+      builder.id(location.name());
       builder.description(location.displayName());
       builder.parent(getOnlyElement(justProvider.get()));
-
       builder.scope(LocationScope.REGION);
-      final Region region = Region.byName(location.name());
+      final Region region = Region.byName(location.displayName());
       if (region != null) {
          builder.iso3166Codes(ImmutableSet.of(region.iso3166Code()));
       }
-
       return builder.build();
    }
 
```

Comment:
Now that we agree we shouldn't change REGION to ZONE, let's discard the changes in this class.